### ISSUE TITLE
User password management & length

### ISSUE THREAD

**Issue by RRAlex** [vmware/harbor#234]
There doesn't seem to be an interface to change the user's password as admin and if you delete the user to recreate it, system will say user exists!
This breaks user management when passwords need to be changed. :(

Secondly, the passwords seem to be limited to a short 20 characters...
Any reason why it couldn't be say 63 or 255?

cheers!

**Comment by reasonerjt** [comment]
@RRAlex 
Currently for audit purpose, when admin delete a user, the user is just "soft deleted", i.e. user with the same username can not be created again.

So you want to enable admin to update a user's password so the username can be re-used?

**Comment by RRAlex** [comment]
So deleting a user would also delete its activity log?

But yes, indeed, simply being able to update a user's password would be sufficient.
This would allow anyone to re-purpose an account: be it to rotate admins creds., correct a mistake, respect password change policy, etc.
And it's much more practical then going into MySQL to delete it manually and re-create / re-attribute the role everywhere, which is management hell and, well, bad UX.

thanks! :-)

**Comment by reasonerjt** [comment]
> So deleting a user would also delete its activity log?

Currently activitly log is associate to user table via a Foreign Key, I know this is debatable, but we may use the term to "Disable" a user, not Delete.

As for enabling admin to update password, the original idea is no one should know any other's password including admin, but it makes sense to me when you mention password change policy, we'll consider implement it.

**Comment by fuhbar** [comment]
@reasonerjt What about the second question @RRAlex raised, the password length? It seems to be restricted to max 20 chars. Is this on purpose? And if so, why?

**Comment by bergtwvd** [comment]
The ability for the admin to reset a user password is **essential** in at least the following situation:
- no LDAP
- no mail server
- user has forgotten his password

But also in the following situation:
- no LDAP
- mail server
- user has changed email address, and forgotten his password

**Comment by tronicum** [comment]
Yeah I would love if we could reset passwords via the web ui..., too

**Comment by bengadbois** [comment]
It would be nice to have the length restriction presented to the user when setting the password

**Comment by phenomax** [comment]
@reasonerjt What about the rather restrictive password length?
